src/DataStructures/DataBox/DataBox.hpp

```diff
@@ -212,13 +212,8 @@ class DataBox<tmpl::list<Tags...>> : private detail::Item<Tags>... {
    * \note the default constructor is only used for serialization
    */
   DataBox() = default;
-  DataBox(DataBox&& rhs) noexcept(
-      tmpl2::flat_all_v<
-          std::is_nothrow_move_constructible_v<detail::Item<Tags>>...>) =
-      default;
-  DataBox& operator=(DataBox&& rhs) noexcept(
-      tmpl2::flat_all_v<
-          std::is_nothrow_move_assignable_v<detail::Item<Tags>>...>) {
+  DataBox(DataBox&& rhs) = default;
+  DataBox& operator=(DataBox&& rhs) {
```

Comment:
I think this is overall a very good change. However, I don't think we should prohibit noexcept specifiers altogether. My suggestion is to follow Google's style guide:

https://google.github.io/styleguide/cppguide.html#noexcept

Reply:
Is it possible to add a git hook that checks for `noexcept`s and prints a warning if you try to commit a change that has them? That way you don't prohibit it altogether, but it's much harder to accidentally add them.

Reply:
> I think this is overall a very good change. However, I don't think we should prohibit noexcept specifiers altogether. My suggestion is to follow Google's style guide:
> 
> https://google.github.io/styleguide/cppguide.html#noexcept

If you are suggesting that I individually examine the 15,437 occurrences of noexcept, that will not be happening...
Note that Google doesn't allow exceptions which is the opposite of what we are moving towards.

I am okay with dropping the file check if we come up with a reasonable policy for when to use noexcept

Reply:
Hehe of course I'm not asking you to check all noexcepts :D My suggestion:

- Drop the commit that prohibits adding noexcept
- In the docs just delete the part on noexcepts. We point to the Google style guide already.
- In the few occurrences where we implement the move constructor and move assignment, keep the noexcept (first few commits).
- Perhaps we should run quick sanity check before / after this change to make sure it doesn't have an unexpected performance hit?

Reply:
@wthrowe @nilsdeppe and I discussed this and our plan is merge this PR as is, and then have a follow-up PR to address these bullets.